src/components/VotingControls/VotingControls.tsx

```diff
@@ -0,0 +1,119 @@
+import React, {useState} from 'react';
+import {ReactComponent as SmileSvg} from '../../assets/icons/smile.svg';
+import {ReactComponent as HeartSvg} from '../../assets/icons/heart.svg';
+import {ReactComponent as SadnessSvg} from '../../assets/icons/sadness.svg';
+import {TypeModal} from '../../constants/constans';
+import {addFavoriteImage, createVote, deleteFavoriteImage} from '../../api/requests';
+import {adaptToServer, feedbackMessage} from '../../utils/utils';
+import VotingModals from './VotingModals/VotingModals';
+import {hideModal, showModal} from '../../store/modalSlice/modalSlice';
+import {useAppDispatch} from '../../hooks';
+import 'react-toastify/dist/ReactToastify.css';
+
+import styles from './VotingControls.module.scss';
+import {Favorites} from '../../types/types';
+
+type VotingControlsProps = {
+  imageId: number | string,
+  isLoaded: boolean,
+  onClick: (nameAction: string | null | undefined) => void,
+  favorites: Favorites[],
+}
+
+function VotingControls({ imageId, isLoaded, onClick, favorites }: VotingControlsProps) {
+  const dispatch = useAppDispatch();
+  const closeModal = () => dispatch(hideModal());
+  const [nameAction, setNameAction] = useState<string | null | undefined>('');
+
+  const postVote = async (payload: number) => {
+    const data = {
+      imageId: imageId,
+      value: payload,
+    };
+
+    try {
+      await createVote(adaptToServer(data));
+      closeModal();
+      onClick(nameAction);
+      feedbackMessage(true);
+    } catch {
+      feedbackMessage(false);
+    }
+  };
+
+  const addFavorites = async () => {
+    try {
+      await addFavoriteImage(adaptToServer({imageId}));
+      closeModal();
+      feedbackMessage(true);
+      onClick(nameAction);
+    } catch (error) {
+      feedbackMessage(false);
+    }
+  };
+
+  const deleteFavorites = async (id: number | string | undefined) => {
+    try {
+      await deleteFavoriteImage({id});
+      closeModal();
+      feedbackMessage(true);
+      onClick(nameAction);
+    } catch (error) {
+      feedbackMessage(false);
+    }
+  };
+
+  const favoriteActions = () => {
+    const getFavoriteId = favorites.find((favorite) => favorite.imageId === imageId);
```

Comment:
В нейминге переменных глагол не следует использовать
```suggestion
    const favoriteItem= favorites.find((favorite) => favorite.imageId === imageId);
```